vpt/processors/speech_detector/__init__.py

```diff
@@ -1,6 +1,83 @@
 '''Processes an audio stream to detect speech.'''
+
+import numpy as np
+from rx.subject import Subject
+from scipy import fft
+
+from vpt.sources.base import SourceBase
 from vpt.processors.base import ProcessorBase
 
+# The frequency range of human voice in Hz.
+VOICE_RANGE = range(80, 1100 + 1)
+# The threshold which the amplitude needs to overcome to be considered present (in undefined units).
+NOISE_THRESHOLD = .5
+SILENCE_THRESHOLD = .3
 
-class SpeechDetector(ProcessorBase):
+
+class SpeechDetector(ProcessorBase[bool]):
     '''Detects speech in an audio stream and outputs a signal stream for that.'''
+    subj = Subject()
```

Comment:
I already wrote this on an earlier PR sometime ago, but this creates a static/class member. All instances of `SpeechDetector` will share the same `subj`. As explained in [PEP 526](https://www.python.org/dev/peps/pep-0526/#class-and-instance-variable-annotations) and [mypy docs](https://mypy.readthedocs.io/en/stable/class_basics.html#instance-and-class-attributes), adding a type annotation will mark the field as an instance attribute, while not annotating implies the old behavior where it is a class attribute. Note that the annotation can still be used inside `__init__`: `self.subj: Observable = Subject()`

Reply:
Also, why isn't it a private field? There is already a getter for it

Reply:
Yeah, but it really doesn't matter in this case, there won't ever be more than one instance of any of these nodes

Reply:
So what? It's not about whether we will have one or more instances, but about the logical consistency. 

Simply, whoever reads the code, if one sees a public field then they assume that it is *intended* to be accessed, modified by external code, and they continue reading with this assumption. If there's no default value, then one may assume that the value should be acquired somewhere else. All of these inconsistencies between the API the code describes and the actual usage negatively affects the readability and may lead to further confusion.

So what? It is about making the life of the one who will read this code next, easier. And this might as well be you.  And especially in this case, when it's very simple to make the code better, we should really go for it.

Reply:
Yeah, fine, it's just that I have to make this change in N more files